test/data/test_dataset.py

```diff
@@ -92,26 +94,61 @@ def test_dataset_inductive_transform():
         -1).all()
 
 
-def test_converter():
+def test_materalization_and_converter():
+    text_embedder_cfg = TextEmbedderConfig(
+        text_embedder=HashTextEmbedder(1),
+        batch_size=8,
+    )
     dataset = FakeDataset(
-        num_rows=10, stypes=[
+        num_rows=10,
+        stypes=[
             stype.categorical,
             stype.numerical,
             stype.multicategorical,
             stype.sequence_numerical,
             stype.timestamp,
+            stype.text_embedded,
             stype.embedding,
-        ]).materialize()
+        ],
+        col_to_text_embedder_cfg=text_embedder_cfg,
+    ).materialize()
+    tf = dataset.tensor_frame
+    text_embedded_col_names = ([
+        col for col, stype in dataset.col_to_stype.items()
+        if stype == stype.text_embedded
+    ])
+    embedding_col_names = ([
+        col for col, stype in dataset.col_to_stype.items()
+        if stype == stype.embedding
+    ])
+    assert len(tf) == len(dataset)
+    assert stype.text_embedded not in tf.feat_dict
+    assert stype.text_embedded not in tf.col_names_dict
+    assert tf.feat_dict[stype.embedding].size(
+        1) == len(text_embedded_col_names) + len(embedding_col_names)
+    assert len(tf.col_names_dict[stype.embedding]
+               ) == len(text_embedded_col_names) + len(embedding_col_names)
+    for col in text_embedded_col_names:
+        assert col in tf.col_names_dict[stype.embedding]
     convert_to_tensor_frame = DataFrameToTensorFrameConverter(
         col_to_stype=dataset.col_to_stype,
         col_stats=dataset.col_stats,
         target_col=dataset.target_col,
         col_to_sep=dataset.col_to_sep,
         col_to_time_format=dataset.col_to_time_format,
+        col_to_text_embedder_cfg=dataset.col_to_text_embedder_cfg,
     )
     tf = convert_to_tensor_frame(dataset.df)
     assert tf.col_names_dict == convert_to_tensor_frame.col_names_dict
     assert len(tf) == len(dataset)
+    assert stype.text_embedded not in tf.feat_dict
+    assert stype.text_embedded not in tf.col_names_dict
+    assert tf.feat_dict[stype.embedding].size(
+        1) == len(text_embedded_col_names) + len(embedding_col_names)
+    assert len(tf.col_names_dict[stype.embedding]
+               ) == len(text_embedded_col_names) + len(embedding_col_names)
+    for col in text_embedded_col_names:
+        assert col in tf.col_names_dict[stype.embedding]
```

Comment:
I think we will have more and more stype.parent testing in the future. Can we re-write this test so that it tests parent grouping in a general way (rather than testing the specific stypes of `embdding` versus `text_embedded`)?

Reply:
The idea is to make this test work off-the-shelf when we add more child-parent stype relationship. I

Reply:
got it!